server/src/systems/missile/cull.rs

```diff
@@ -13,25 +13,22 @@ pub struct MissileCull;
 #[derive(SystemData)]
 pub struct MissileCullData<'a> {
 	pub ents: Entities<'a>,
-	pub spawntime: ReadStorage<'a, MobSpawnTime>,
+	pub missile_trajectory: ReadStorage<'a, MissileTrajectory>,
+	pub pos: ReadStorage<'a, Position>,
 	pub mob: ReadStorage<'a, Mob>,
-	pub config: Read<'a, Config>,
-	pub thisframe: Read<'a, ThisFrame>,
 	pub conns: Read<'a, Connections>,
 }
 
 impl<'a> System<'a> for MissileCull {
 	type SystemData = MissileCullData<'a>;
 
 	fn run(&mut self, data: MissileCullData<'a>) {
-		(&*data.ents, &data.mob, &data.spawntime)
+		(&*data.ents, &data.mob, &data.pos, &data.missile_trajectory)
 			.join()
-			.filter_map(|(ent, mob, spawntime)| {
-				let ref info = data.config.mobs[*mob];
-
-				let dt = data.thisframe.0 - spawntime.0;
-
-				if dt > info.lifetime {
+			.filter_map(|(ent, mob, pos, missile_trajectory)| {
+				let distance_traveled = (*pos - missile_trajectory.0).length().inner();
+				let end_distance = missile_trajectory.1.inner();
```

Comment:
You don't need the `.inner()` here since both `distance_traveled` and `end_distance` have a type of `Distance`.

I think that doing the following would be cleaner.
```rust
let distance_traveled = (*pos - missile_trajectory.0).length();
let end_distance = missile_trajectory.1;
```

Reply:
Right! I wanted to do that but it wouldn't compile. Must've been tired.